components/speedreader/rust/lib/src/readability/src/extractor.rs

```diff
@@ -123,8 +310,11 @@ pub fn extract_dom<S: ::std::hash::BuildHasher>(
     post_process(&mut dom, top_candidate.clone(), &meta);
 
     // Calls html5ever::serialize() with IncludeNode for us.
-    let charset_blob = format!("<meta charset=\"{}\"/>", meta.charset);
-    let content: String = charset_blob + &top_candidate.to_string();
+    let mut content: String = top_candidate.to_string();
+    if let Some(ref charset) = meta.charset {
```

Comment:
im not sure i understand the "why" for this part.  Can you explain a bit, either here or w/ a comment?

Reply:
the browsers default to ascii encoding if none is specified, so i pull the charset from the `meta` tags because when we choose a new body, the entire head gets removed. so i need to add that back in.

Reply:
follow up issue regarding encoding: https://github.com/brave/brave-browser/issues/15518